lib/drivers/tronscan.js

```diff
@@ -0,0 +1,148 @@
+const Driver = require('../models/driver');
+const Supply = require('../models/supply');
+const { promisesMap } = require('../util');
+const SupplyModifier = require('../models/supply-modifier');
+const Coin = require('../models/coin');
+
+/**
+ * Tronscan driver. Supports
+ * total and circulating supply
+ * for coin TRX and token XDCA on
+ * Tronscan blockchain
+ *
+ *
+ * @memberof Driver
+ * @augments Driver
+ */
+class Tronscan extends Driver {
+  constructor(options) {
+    super({
+      timeout: 100, // 10 requests per second
+      supports: {
+        circulating: true,
+        blockchains: ['Tron'],
+      },
+      options,
+    });
+  }
+
+  /** fetch total supply of native token
+   *
+   * @augments Driver.fetchTotalSupply
+   * @async
+   */
+  async fetchTotalSupply() {
+    const { data } = await this.request({
+      url: 'https://apilist.tronscan.org/api/system/proxy',
+      method: 'post',
+      body: {
+        url: 'https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/latest?symbol=TRX&convert=BTC',
+      },
+    });
```

Comment:
I don't really like to have CoinMarketCap here as a source for our supply data. It doesn't fit in this driver and it makes it quite non-transparent.
To get the total supply from Tronscan we could use this endpoint: https://apilist.tronscan.org/api/funds.
Where we can calculate the total supply like they explain on this page https://tronscan.org/#/data/stats/supply.
Which should be like this:
```
+ genesisBlockIssue
+ totalBlockPay
+ totalNodePay
- burnByCharge
- burnPerDay
-----------------
= total supply
```